src/line.rs

```diff
@@ -96,33 +97,113 @@ impl<'input, Endian> From<&'input [u8]> for DebugLine<'input, Endian>
     }
 }
 
+/// A `LineNumberProgramHolder` provides access to a `LineNumberProgramHeader` and
+/// a way to add files to the files table if necessary. Gimli consumers should
+/// never need to use or see this trait.
+pub trait LineNumberProgramHolder<'input, Endian>
+    where Endian: Endianity
+{
+    /// Get a reference to the held `LineNumberProgramHeader`.
+    fn header<'a>(&'a self) -> &'a LineNumberProgramHeader<'input, Endian>;
+    /// Add a file to the file table if necessary.
+    fn add_file(&mut self, file: FileEntry<'input>);
+}
+
+/// `IncompleteLineNumberProgramHolder` holds a `LineNumberProgramHeader` that needs
+/// files defined by the line number program header to be added to it. Gimli consumers
+/// should never need to use or see this trait.
+#[derive(Debug)]
+pub struct IncompleteLineNumberProgramHolder<'input, Endian>
```

Comment:
Delete this struct, it adds no benefit. Replace its usage with `IncompleteLineNumberProgram`.